### PR TITLE
feat(route): add 豆瓣网站中关于某书的论坛页面(#13838)

### LINKED ISSUE #13838: 豆瓣读书 关于某书的论坛页面  请求支持路由
### 类型

社交媒体

### 网站地址

https://book.douban.com/subject/36328704/discussion/

### 网站描述

豆瓣读书子页面

### 需要生成什么内容？

生成提示网页链接对应页面的路由

### 额外描述

无

### 这不是重复的 RSS 请求

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues) 和 [pull requests](https://github.com/DIYgod/RSSHub/pulls)，以确保该 RSS 尚未被请求。

### PR DESCRIPTION
**Pull request by nightmare-mio**
## Involved Issue / 该 PR 相关 Issue

Close #13838

## Example for the Proposed Route(s) / 路由地址示例

```routes
/douban/36328704/discussion
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [v2 Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [v2 路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Documentation / 文档说明
  - [ ] EN / 英文文档
  - [ ] CN / 中文文档
- [x] Full text / 全文获取
  - [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制 
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施? 
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
:id为书本id，可以从url中获取，仅获取论坛第一页文章与内容。

### CODE REVIEW

File: lib/v2/douban/other/discussion.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+
+module.exports = async (ctx) => {
+    const { id } = ctx.params;
+    const link = 'https://book.douban.com/subject';
+    const { data: response } = await got({
+        method: 'get',
+        url: `${link}/${id}/discussion`,
```
Comment (TonyRL) [inline lib/v2/douban/other/discussion.js:16]: Use the redirected url 
https://github.com/DIYgod/RSSHub/actions/runs/6945849477/job/18896148171#step:12:17
```suggestion
        url: `${link}/${id}/discussion/`,
```

File: lib/v2/douban/other/discussion.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            item = $(item);
+            const a = item.find('a').first();
+            return {
+                title: a.attr('title'),
+                link: a.attr('href'),
+                pubDate: parseDate(item.find('.time').text()),
+                author: item.find('a').eq(1).text(),
+            };
+        });
+    list.shift();
```
Comment (TonyRL) [inline lib/v2/douban/other/discussion.js:33]: You can use `.slice(1)` before `.map()`

File: lib/v2/douban/other/discussion.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+
+module.exports = async (ctx) => {
+    const { id } = ctx.params;
+    const link = 'https://book.douban.com/subject';
+    const { data: response } = await got({
+        method: 'get',
+        url: `${link}/${id}/discussion`,
+    });
```
Comment (TonyRL) [inline lib/v2/douban/other/discussion.js:17]: Please use either one style but not both.
https://github.com/DIYgod/RSSHub/blob/d83b6e375844fbca21fff7feebf621b5f7693699/lib/v2/douban/other/discussion.js#L39

File: lib/v2/douban/other/discussion.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                const $ = cheerio.load(response);
+                // 评论
+                const list = $('#comments>.comment-item').toArray();
+                let no = 0;
+                const replyContent = list
+                    .map((item) => {
+                        const post = $(item);
+                        const content = post.find('.content>p').html();
+                        const author = post.find('.author>a').text();
+                        no++;
```
Comment (TonyRL) [inline lib/v2/douban/other/discussion.js:49]: The second parameter of [`.map()`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/map) is the index. You don't need a `no` to keep track of the index

**Review by TonyRL — COMMENTED** [review]
Missing corresonding docs, maintainer and radar entries.

File: lib/v2/douban/other/discussion.js (hunk 4 above)
Comment (nightmare-mio) [reply lib/v2/douban/other/discussion.js:45]: Unable to distinguish between article content and comments if without no

Comment (TonyRL) [reply lib/v2/douban/other/discussion.js:45]: You can still distinguish between article content and comments.

> The second parameter of [`.map()`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/map) is the index. You don't need a `no` to keep track of the index

```js
                const replyContent = list
                    .map((item, index) => {
                        const post = $(item);
                        const content = post.find('.content>p').html();
                        const author = post.find('.author>a').text();
                        return `<p><div>#${index + 1}: <i>${author}</i></div><div>${content}</div></p>`;
                    })
                    .join('');
```

Comment (nightmare-mio) [reply lib/v2/douban/other/discussion.js:45]: # don't have no
QQ截图20231122212544
# have no,like v2ex
QQ截图20231122212638

Comment (nightmare-mio) [reply lib/v2/douban/other/discussion.js]: thanks

**Review by TonyRL — COMMENTED** [review]
Missing corresonding [maintainer](https://docs.rsshub.app/joinus/new-rss/before-start#maintainerjs) and [radar](https://docs.rsshub.app/joinus/new-radar) entries.

**Comment by nightmare-mio** [comment]
> 缺少相应的[维护人员](https://docs.rsshub.app/joinus/new-rss/before-start#maintainerjs)和[雷达](https://docs.rsshub.app/joinus/new-radar)边界。
I finished。sorry，I don‘t have Pull Request EXP。

**Pull request merged**